src/charm.py

```diff
@@ -141,6 +142,12 @@ def func_with_catch_errors(self: "GithubRunnerCharm", event: EventT) -> None:
         except MissingMongoDBError as err:
             logger.exception("Missing integration data")
             self.unit.status = WaitingStatus(str(err))
+        except IntegrationNotFoundError as err:
+            logger.exception("Missing integration")
+            self.unit.status = BlockedStatus(str(err))
+        except IntegrationDataNotReadyError as err:
+            logger.exception("Missing integration data")
```

Comment:
This is the same msg as in line 143. Is this intentional?

Reply:
I think @gregory-schiano did not like the abstract name (I also originally had "MissingIntegrationDataError" but then changed it to `MissingMongoDBError`). I am fine with using the abstract name and merging this with lines 142-144, but perhaps confirm with @gregory-schiano .